### DISCUSSION TITLE
Input can overwrite previous output

### DISCUSSION THREAD

**Discussion by tristan957** [ghostty-org/ghostty#2783] [Q&A] [answered]
When I run `zig build`, and then cancel it, Ghostty places the cursor at a position which was previously output. It took me a while to find something that reliably reproduces this. I have previously experienced this with aerc and neovim too.

https://github.com/user-attachments/assets/1ccef60f-eace-4d41-a2b8-8ebc8ab591f1

**Comment by mitchellh** [comment]
I don’t think this is a terminal issue. Zig needs to move the cursor on interrupt or the shell needs to do something.

Neovim and aerc shouldn’t be affected since I assume they use alt screen with automatic cursor restore. If there’s a way to break that I’d be curious.

**Reply by tristan957** [reply]
I don't know how to produce it with aerc or neovim, but it happens to me pretty much daily.

**Reply by rhodes-b** [reply]
Can confirm I just got this @tristan957 when closing neovim reading some ghostty source, first time it happened to me. I have also seen the zig one

**Reply by rhodes-b** [reply]
![image](https://github.com/user-attachments/assets/1006a874-3ab3-4f38-ba49-99a4645da251)

can see `nvim .` on the bottom where I launched the repo

**Comment by tristan957** [comment]
I was able to reproduce this with just `less(1)`. Again, I don't know how to reliably reproduce unfortunately.

**Comment by 00-kat** [comment]
I haven't seen this in Ghostty yet, but I remember seeing this happen randomly in Kitty back when I used it every once in a while when exiting my editor too.  I'm not sure if I had resized my window between opening and closing my editor though.  I'm not sure if this is an issue with Ghostty itself, Ghostty's shell integration (which might also be why it happened in Kitty to me), or some other obscure issue with the TUIs in question itself.

**Reply by tristan957** [reply]
I think you might be a genius. I found a reproducer. Start Ghostty non-maximized. Run echo enough times that you have more output than the size of the window. Start Neovim. Maximize Ghostty. Quit Neovim.

Video is hosted on my Fastmail account: https://user.fm/files/v2-155dda3fb7242f34f87be45855288a16/Screencast%20From%202025-02-09%2010-33-55.mp4

**Reply by tristan957** [reply]
It seems like Ghostty loses track of the cursor position or something.

**Reply by 00-kat** [reply]
I can reproduce that too, I lost four lines of scrollback once Neovim exited.

Edit: four is how much maximising changes the vertical size of the window for me[^1], according to the resize overlay.

[^1]: My gaps are pretty large...

**Reply by 00-kat** [reply]
> Ghostty's shell integration

I can reproduce it with shell integration disabled as well.  I'm now guessing Kitty also had this bug, rather than the shell integration being problematic, but later fixed it or something (I couldn't reproduce it with either resizing or maximising in Kitty when I just tried now).

**Comment by mitchellh** [comment, marked as answer]
#5718
